musicbot/bot.py

```diff
@@ -388,7 +388,8 @@ def _delete_old_audiocache(self, path=AUDIO_CACHE_PATH):
 
                 async for lmsg in self.logs_from(channel, limit=1):
                     if lmsg != last_np_msg and last_np_msg:
-                        await self.safe_delete_message(last_np_msg)
+                        if self.config.delete_nowplaying:
```

Comment:
I'd suggest putting this check above line 387, otherwise the bot gets the logs from the channel first before even checking if it will do anything with them